app/value_objects/value_object.rb

```diff
@@ -21,13 +21,19 @@ def hash
   end
 
   class << self
+    # Define inquiry methods for each value in the value object,
+    # i.e. `#coffee?` returns true for value `:coffee`, false for `:tea`
     def inherited(subclass)
-      TracePoint.trace(:end) do
-        # Define inquiry methods for each value in the value object,
-        # i.e. `#coffee?` returns true for value `:coffee`, false for `:tea`
+      TracePoint.trace(:end) do |trace|
         # :nocov:
-        subclass.values.each do |value|
-          define_method("#{value}?") { value.eql?(self) }
+        if subclass == trace.self
+          subclass.const_set(
+            :INQUIRY_METHODS, subclass.values.map { |value| "#{value}?".to_sym }
```

Comment:
Why do you set constants as well as methods?

Reply:
So the constant will return this (for ApplicationStatus, but other value-objects behave the same):

```ruby
> ApplicationStatus::INQUIRY_METHODS
=> [:in_progress?, :submitted?, :returned?, :superseded?]
```

And then we have those instance methods declared and returning true/false depending if the value is what you are asking it for. Example:

```ruby
> s = ApplicationStatus.new(:returned)
> s.returned?
=> true
> s.submitted?
=> false
> s.in_progress?
=> false
> s.superseded?
=> false
> s.foobar?
(irb):8:in `<main>': undefined method `foobar?' for #<ApplicationStatus:0x000000010dee1980 @value=:returned> (NoMethodError)
```

Before, the constant was no needed, but I've now found a use case where it came handy to have an array of all the inquiry methods for a given value-object. In this case it was to perform delegation of all those methods (and not having to add one by one, or risk forgetting if we ever add another status, etc.)